ios/FluentUI/Vnext/Button/ButtonTokens.swift

```diff
@@ -30,145 +30,291 @@ import UIKit
-
-    @Published public var titleColor: UIColor!
-    @Published public var borderColor: UIColor!
-    @Published public var backgroundColor: UIColor!
-    @Published public var iconColor: UIColor!
-
-    @Published public var highlightedTitleColor: UIColor!
-    @Published public var highlightedBorderColor: UIColor!
-    @Published public var highlightedBackgroundColor: UIColor!
-    @Published public var highlightedIconColor: UIColor!
-
-    @Published public var disabledTitleColor: UIColor!
-    @Published public var disabledBorderColor: UIColor!
-    @Published public var disabledBackgroundColor: UIColor!
-    @Published public var disabledIconColor: UIColor!
-
-    @Published public var restShadow1Color: Color!
-    @Published public var restShadow1Blur: CGFloat!
-    @Published public var restShadow1DepthX: CGFloat!
-    @Published public var restShadow1DepthY: CGFloat!
-    @Published public var restShadow2Color: Color!
-    @Published public var restShadow2Blur: CGFloat!
-    @Published public var restShadow2DepthX: CGFloat!
-    @Published public var restShadow2DepthY: CGFloat!
-
-    @Published public var pressedShadow1Color: Color!
-    @Published public var pressedShadow1Blur: CGFloat!
-    @Published public var pressedShadow1DepthX: CGFloat!
-    @Published public var pressedShadow1DepthY: CGFloat!
-    @Published public var pressedShadow2Color: Color!
-    @Published public var pressedShadow2Blur: CGFloat!
-    @Published public var pressedShadow2DepthX: CGFloat!
-    @Published public var pressedShadow2DepthY: CGFloat!
-
-    var style: MSFButtonStyle
-    var size: MSFButtonSize
-
-    init(style: MSFButtonStyle,
-         size: MSFButtonSize) {
+public class ButtonTokens: ControlTokens {
+    public init(style: MSFButtonStyle,
+                size: MSFButtonSize) {
         self.style = style
         self.size = size
 
         super.init()
-
-        self.themeAware = true
-        updateForCurrentTheme()
     }
 
-    @objc open func didChangeAppearanceProxy() {
-        updateForCurrentTheme()
-    }
+    let style: MSFButtonStyle
+    let size: MSFButtonSize
 
-    override func updateForCurrentTheme() {
-        let currentTheme = theme
-        var appearanceProxy: AppearanceProxyType
+    lazy var borderRadius: CGFloat = {
+        switch size {
+        case .small:
+            return globalTokens.borderRadius[.large]
+        case .medium:
+            return globalTokens.borderRadius[.large]
+        case .large:
+            return globalTokens.borderRadius[.xLarge]
+        }
+    }()
 
+    lazy var borderSize: CGFloat = {
         switch style {
-        case .primary:
-            appearanceProxy = currentTheme.MSFPrimaryButtonTokens
         case .secondary:
-            appearanceProxy = currentTheme.MSFSecondaryButtonTokens
-        case .ghost:
-            appearanceProxy = currentTheme.MSFGhostButtonTokens
+            return globalTokens.borderSize[.thin]
+        default:
```

Comment:
I know in devmain (or at least the FRE code) we don't like defaults, but I'm not sure how we feel about them here. Probably similarly, so it might be better to just list out all the options?

Reply:
Interesting question. In this case, I was looking at the model of the old `MSFButtonTokens` YAML file, where the basic structure is

```base tokens
```    |
```    |----> override 1
```    |----> override 2
```    |----> override 3

So there really is a "default" set that is automatically inherited by all control types, unless explicitly overridden, and using the `default` case helps to explicitly distinguish that.

That said, `Button` is unique among Fluent controls, as far as I can tell, in that no control style actually uses the "base" values as-is, which makes parsing out what value to use where a bit challenging. Perhaps that's also something worth investigating at some point?

Reply:
Seems like it might be worth revisiting some other time to separate out the defaults, for many of the same reasons you gave in the comment about combining the .small and .medium above.

Reply:
I believe Swiftlint has a check on switch statements that require them to be exaustive.
It's definitely a good practice in devmain code that no default is used to avoid silent bugs when new items in the enumeration are introduced.
I think we should not introduce defaults in the code and work on removing the remaining ones throughout.
Quick note: there are some enumerations named "default", which in this case is fine although a bit confusing:

`case .default:`